src/main/java/duke/AddTaskCommand.java

```diff
@@ -0,0 +1,93 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+public class AddTaskCommand extends Command {
+
+    public AddTaskCommand(String command, String input, TaskList taskList) {
+        super(command, input, taskList);
+    }
+
+    /**
+     * Checks if the user input is formatted into a correct To-do command.
+     * If it is, it adds the To-do to the task list and prints the Add task message.
+     * Otherwise, it prints the exception faced.
+     */
+    private void addTodo() {
+        try {
+            if (parser.canParseTodo(input)) {
+                String description = parser.parseTodoDescription(input);
+                Task task = new Todo(description);
+                taskList.addTask(task);
+                ui.printAddTask(task, taskList.getSize());
+            } else {
+                throw new WrongFormatDukeException(command);
+            }
+        } catch (DukeException e) {
+            ui.printError(e);
+        }
+    }
+
+    /**
+     * Checks if the user input is formatted into a correct Deadline command.
+     * If it is, it adds the Deadline to the task list and prints the Add task message.
+     * Otherwise, it prints the exception faced.
+     */
+    private void addDeadline() {
+        try {
+            if (parser.canParseDeadline(input)) {
+                try {
+                    LocalDateTime dateTime = parser.parseDeadlineDate(input);
+                    String description = parser.parseDeadlineDescription(input);
+                    Deadline task = new Deadline(dateTime, description);
+                    taskList.addTask(task);
+                    ui.printAddTask(task, taskList.getSize());
+                } catch (DateTimeParseException e) {
+                    ui.printError(new WrongFormatDukeException(command));
+                }
```

Comment:
This try-catch section is quite deeply nested. Perhaps it could be avoided by catching the `DateTimeParseException` together with the `DukeException` in the catch clause below.